x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticQueryBuilder.java

```diff
@@ -171,11 +179,12 @@ private QueryBuilder doRewriteBuildSemanticQuery(SearchExecutionContext searchEx
             }
 
             return semanticTextFieldType.semanticQuery(inferenceResults, searchExecutionContext.requestSize(), boost(), queryName());
-        } else {
+        } else if (throwOnUnsupportedField) {
             throw new IllegalArgumentException(
                 "Field [" + fieldName + "] of type [" + fieldType.typeName() + "] does not support " + NAME + " queries"
             );
         }
+        return this;
```

Comment:
If we need to keep `throwOnUnsupportedField`, should we return `MatchNoneQueryBuilder` here?

Reply:
I don't think we want to do that. 

Basically, if `throwOnUnsupportedField` is true, it defaults to today's behavior: 
```
{
  "took": 126,
  "timed_out": false,
  "_shards": {
    "total": 2,
    "successful": 1,
    "skipped": 0,
    "failed": 1,
    "failures": [
      {
        "shard": 0,
        "index": "my-index-2",
        "node": "HtyL9wCRTtSGKf-aXuS89g",
        "reason": {
          "type": "illegal_argument_exception",
          "reason": "Field [inference_field] of type [text] does not support semantic queries"
        }
      }
    ]
  },
  "hits": {
    "total": {
      "value": 1,
      "relation": "eq"
    },
    "max_score": 12.725867,
    "hits": [
      {
        "_index": "my-index-1",
        "_id": "1",
        "_score": 12.725867,
        "_source": {
          "inference_field": {
            "text": "a test value",
            "inference": {
              "inference_id": "my-elser-endpoint",
              "model_settings": {
                "task_type": "sparse_embedding"
              },
              "chunks": [
                {
                  "text": "a test value",
                  "embeddings": {
                    "test": 2.9745526,
                    "value": 2.7412868,
                    "testing": 1.9397259,
                    "tests": 1.3399959,
                    "tested": 1.1991253,
                    "values": 1.1034975,
                    "trial": 0.896227,
                    "measure": 0.8546149,
                    "gage": 0.74623615,
                    "a": 0.7101594,
                    "analysis": 0.638586,
                    "term": 0.61701477,
                    "result": 0.5063082,
                    "assessment": 0.50300145,
                    "index": 0.50018245,
                    "definition": 0.45780453,
                    "accounting": 0.44581136,
                    "quality": 0.4282373,
                    "performance": 0.41900125,
                    "##test": 0.38732845,
                    "unit": 0.34403202,
                    "exam": 0.32937595,
                    "symbol": 0.3075734,
                    "assess": 0.28477323,
                    "sample": 0.2844404,
                    "precision": 0.28208783,
                    "quantity": 0.24788333,
                    "ass": 0.24601695,
                    "math": 0.24055041,
                    "q": 0.24032095,
                    "measurement": 0.2231896,
                    "calculation": 0.22096556,
                    "score": 0.2020141,
                    "error": 0.17952153,
                    "meaning": 0.17108563,
                    "report": 0.16626503,
                    "lab": 0.14854811,
                    "significance": 0.12679343,
                    "statistical": 0.102960005,
                    "question": 0.10062234,
                    "percent": 0.091963604,
                    "formula": 0.089570634,
                    "quantitative": 0.054434557,
                    "table": 0.040091787,
                    "display": 0.036803197,
                    "qualification": 0.033828333,
                    "experiment": 0.008009695,
                    "inventory": 0.0077431058,
                    "sy": 0.0054949666,
                    "evidence": 0.0021653557
                  }
                }
              ]
            }
          },
          "text_field": "a test value"
        }
      }
    ]
  }
}
```

If false, we still return the 1 result, but we don't return the error. This is parity with how it works today.

Reply:
Understood that we want `throwOnUnsupportedField` to be false in this case. What I'm saying is, if `throwOnUnsupportedField` is `false`, we get to this line. Would it be better to return `MatchNoneQueryBuilder` here? It means that we are querying a non-semantic text field, so it's pointless to continue rewriting.

Reply:
Addressed in b09e1d0a4ab98beb217ba3a7ff5d3270b1012231